Windows bug: SW bundle fails with 'Relative import path not prefixed' when templates are embedded

## Summary

When bundling a Service Worker app that has `templatesDir` configured, the bundler creates a virtual entry file. On Windows, the deno-resolver plugin fails to resolve relative imports from this virtual entry, producing a misleading error.

## Error message

```
X [ERROR] Relative import path "./src/farmhub-sw/sw.ts" not prefixed with / or ./ or ../
and not in import map from "alexi-sw-entry-virtual:C:/Users/Atzu/Code/farmhub/__alexi_sw_entry__.ts"
[plugin deno-resolver]

  alexi-sw-entry-virtual:C:/Users/Atzu/Code/farmhub/__alexi_sw_entry__.ts:2:14:
    2 │ export * from "./src/farmhub-sw/sw.ts";
      ╵               ~~~~~~~~~~~~~~~~~~~~~~~~
```

## Root cause

In `@alexi/staticfiles/commands/bundle.ts`, when `templates.length > 0`, the bundler creates a virtual entry and sets its namespace to `"alexi-sw-entry-virtual"` and its path to the absolute Windows path of the CWD:

```ts
const cwdNorm = Deno.cwd().replace(/\/g, "/");
const virtualEntryPath = `${cwdNorm}/__alexi_sw_entry__.ts`;
// → "C:/Users/Atzu/Code/farmhub/__alexi_sw_entry__.ts"
```

The `@luca/esbuild-deno-loader` deno-resolver plugin then constructs a referrer URL for resolving imports from that virtual entry:

```ts
// plugin_deno_resolver.ts
if (args.importer !== "") {
  referrer = new URL(`${args.namespace}:${args.importer}`);
  // → new URL("alexi-sw-entry-virtual:C:/Users/Atzu/Code/farmhub/__alexi_sw_entry__.ts")
}
```

On **Linux/macOS** this works because `args.importer` is `/home/...` — the resulting URL `alexi-sw-entry-virtual:/home/...` is valid.

On **Windows** `args.importer` is `C:/Users/...` — the `C:` is interpreted as a port or part of the scheme, making the URL invalid. The resolver then falls back to treating `"./src/farmhub-sw/sw.ts"` as a bare specifier and fails.

Note: `onLoad` correctly sets `resolveDir: cwdNorm`, but deno-resolver only uses `resolveDir` when `args.importer === ""`, which never happens here.

## Affected versions

- `@alexi/staticfiles@0.26.1`
- `@luca/esbuild-deno-loader@0.11.1` (indirect — the URL construction is in this plugin, but it is fed a Windows path by the alexi bundler)

## Reproduction

1. Windows machine
2. Alexi SW app with `templatesDir` set in `app.ts`
3. Run `deno task manage bundle --settings <sw-settings>`

## Suggested fix

The simplest fix is to not use the absolute CWD path as the virtual entry path. Using only the filename and relying on `resolveDir` avoids the Windows path in the namespace URL entirely:

```ts
// commands/bundle.ts — current
const cwdNorm = Deno.cwd().replace(/\/g, "/");
const virtualEntryPath = `${cwdNorm}/__alexi_sw_entry__.ts`;

// fix option A: use filename only, resolveDir handles location
const virtualEntryPath = `__alexi_sw_entry__.ts`;
```

Alternatively, convert the CWD to a `file://` URL so the namespace+importer combination always produces a valid URL regardless of OS:

```ts
// fix option B: use file:// URL as the path
import { toFileUrl } from "jsr:@std/path";
const virtualEntryPath = toFileUrl(`${Deno.cwd()}/__alexi_sw_entry__.ts`).href;
// → "file:///C:/Users/Atzu/Code/farmhub/__alexi_sw_entry__.ts"
```

With option B the deno-resolver would construct:
```ts
new URL("alexi-sw-entry-virtual:file:///C:/Users/...")  // still invalid
```

So option A (bare filename + resolveDir) is cleaner. The `onResolve` filter regex already matches `__alexi_sw_entry__.ts` by name, so no other change is needed.

--- a/src/staticfiles/commands/bundle.ts
+++ b/src/staticfiles/commands/bundle.ts
@@ -765,9 +765,17 @@ export class BundleCommand extends BaseCommand {
         JSON.stringify("./" + entryPoint.replace(/^\.\//, ""))
       };\n`;
 
-      // Use an absolute path for the virtual entry so relative imports resolve
+      // Use only the filename as the virtual entry path so that the namespace +
+      // path combination is always a valid URL on every OS.  On Windows an
+      // absolute path such as "C:/Users/..." would be interpreted as an invalid
+      // port/scheme by @luca/esbuild-deno-loader when it constructs:
+      //   new URL("alexi-sw-entry-virtual:C:/Users/...")
+      // which breaks relative-import resolution.  Using just the filename avoids
+      // this: the deno-resolver plugin uses `resolveDir` (set below) to resolve
+      // imports from the virtual entry instead of deriving a base URL from the
+      // importer path.  See https://github.com/atzufuki/alexi/issues/172.
       const cwdNorm = Deno.cwd().replace(/\\/g, "/");
-      const virtualEntryPath = `${cwdNorm}/__alexi_sw_entry__.ts`;
+      const virtualEntryPath = "__alexi_sw_entry__.ts";
       const virtualEntryNamespace = "alexi-sw-entry-virtual";
 
       // Add a plugin that intercepts the virtual entry path

